app/src/main/java/com/example/demomvp/screen/adapter/MainAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.demomvp.screen.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.demomvp.R
+import com.example.demomvp.data.model.Champion
+import com.example.demomvp.utils.LoadImageBitmap
+import com.example.demomvp.utils.OnItemRecyclerViewClickListener
+import kotlinx.android.synthetic.main.item_layout_champion.view.*
+
+class MainAdapter : RecyclerView.Adapter<MainAdapter.ViewHolder?>() {
+
+    private val champions = mutableListOf<Champion>()
+    private var onItemClickListener: OnItemRecyclerViewClickListener<Champion>? = null
+
+    fun updateData(champions: MutableList<Champion>?) {
+        champions?.let {
+            this.champions.clear()
+            this.champions.addAll(champions)
```

Comment:
```suggestion
            champions.clear()
           champions.addAll(it)
```

Reply:
Cái this.champions này là tham chiếu đến cái list champions được khai báo bên ngoài anh. Còn cái champions này là cái tham số truyền vào của hàm ạ